docs/spec_draft.md

```diff
@@ -680,6 +681,48 @@ IEEE-754 specification.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.if
+
+### Semantics
+
+Returns the result of `true_branch` if `pred` is `true`, `false_branch` if
+`pred` is `false`.
+
+### Inputs
+
+| Name           | Type                                       |
+|----------------|--------------------------------------------|
+| `pred`         | 1-dimensional tensor constant of type `i1` |
+| `true_branch`  | function of type `function`                |
+| `false_branch` | function of type `function`                |
+
+### Outputs
+
+| Name     | Type               |
+|----------|--------------------|
+| `result` | any supported type |
+
+### Constraints
+
+  * (C1) `true_branch` and `false_branch` take 0 arguments.
+  * (C2) Return types of `true_branch` and `false_branch` are same.
```

Comment:
This also needs "(C3) For all `i`, `type(results[i])` = `type(true_branch).outputs[i]`.